crates/hyfetch/src/pride_month.rs

```diff
@@ -0,0 +1,188 @@
+
+        // Loop over the height
+        for y in 0..h {
+            // Print the starting color
+            buf += &colors[((frame + usize::from(y)) / usize::from(block_width)) % colors.len()]
+                .to_ansi_string(color_mode, ForegroundBackground::Background);
+            buf += &fg.to_ansi_string(color_mode, ForegroundBackground::Foreground);
+
+            // Loop over the width
+            for x in 0..w {
+                let idx = frame
+                    + usize::from(x)
+                    + usize::from(y)
+                    + (2.0 * (y as f64 + 0.5 * frame as f64).sin()) as usize;
+                let y_text = text_start_y <= y && y < text_end_y;
+
+                let border = 1 + if y == text_start_y || y == text_end_y - 1 {
+                    0
+                } else {
+                    1
+                };
+
+                // If it's a switching point
+                if idx % usize::from(block_width) == 0
+                    || x == text_start_x - border
+                    || x == text_end_x + border
+                    || x == notice_start_x - 1
+                    || x == notice_end_x + 1
+                {
+                    // Print the color at the current frame
+                    let c = colors[(idx / usize::from(block_width)) % colors.len()];
+                    if (y_text && (text_start_x - border <= x) && (x < text_end_x + border))
+                        || (y == notice_y && notice_start_x - 1 <= x && x < notice_end_x + 1)
+                    {
+                        let c: LinSrgba = c.with_alpha(1.0).into_linear();
+                        let c = Srgb::<u8>::from_linear(c.overlay(black).without_alpha());
+                        buf += &c.to_ansi_string(color_mode, ForegroundBackground::Background);
+                    } else {
+                        buf += &c
+                            .into_format()
+                            .to_ansi_string(color_mode, ForegroundBackground::Background);
+                    }
+                }
+
+                // If text should be printed, print text
+                if y_text && text_start_x <= x && x < text_end_x {
+                    buf.push(
+                        text_lines[usize::from(y - text_start_y)]
+                            .chars()
+                            .nth(usize::from(x - text_start_x))
+                            .unwrap(),
+                    );
+                } else if y == notice_y && notice_start_x <= x && x < notice_end_x {
+                    buf.push(NOTICE.chars().nth(usize::from(x - notice_start_x)).unwrap());
+                } else {
+                    buf.push(' ');
+                }
+            }
+
+            // New line if it isn't the last line
+            if y != h - 1 {
+                buf += &color("&r\n", color_mode)
+                    .expect("line separator should not contain invalid color codes");
+            }
+        }
+
+        write!(io::stdout(), "{buf}")
```

Comment:
We suffer from tearing. This would probably help avoid tearing: https://docs.rs/crossterm/latest/crossterm/terminal/struct.BeginSynchronizedUpdate.html